src/hooks/useEventEmitter.ts

```diff
@@ -94,3 +133,18 @@ export function useEventEmitterState<T>(
     useEventEmitter(emitter, eventName, handler);
     return value;
 }
+
+export function useEventTargetState<T>(
```

Comment:
Seems a bit strange to use an overload on one but a separate method on the other, no?

Reply:
The idea was to avoid duplicating the code though I can split it up, if you think it's better

Reply:
I don't mind either, but can we use the same approach for both hooks?